server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -701,7 +702,8 @@ private static IndexNotFoundException indexNotFoundException(String expression)
                 if (context.getOptions().ignoreAliases()) {
                     return metaData.getAliasAndIndexLookup().entrySet().stream()
                             .filter(e -> e.getValue().isAlias() == false)
-                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
+                                (v1, v2) -> {throw new IllegalStateException("no duplicate indices should exist");}, TreeMap::new));
```

Comment:
out of curiosity - why did you add this?

Reply:
oh woops. this was not meant to stay. I added this to debug where the ordering of indices was being lost. will revert